redis.go

```diff
@@ -9,58 +9,85 @@ import (
 	rediscache "github.com/go-redis/redis/v8"
 )
 
+var (
+	_ Cache    = (*redisCache)(nil)
+	_ CacheCtx = (*redisCache)(nil)
+)
+
 type redisCache struct {
 	ctx    context.Context
 	client *rediscache.Client
 	ttl    time.Duration
-	Cache
 }
 
-func NewRedisCache(ctx context.Context, cacheURL string, ttl time.Duration) (Cache, error) {
+func NewRedisCacheWithClient(client *rediscache.Client, ttl time.Duration) *redisCache {
+	return &redisCache{
+		ctx:    context.Background(),
+		client: client,
+		ttl:    ttl,
+	}
+}
+
+func NewRedisCache(ctx context.Context, cacheURL string, ttl time.Duration) (*redisCache, error) {
```

Comment:
both structs with contexts as fields and constructor accepting a context is usually a sign that is doing something that probably is better done outside. 

It's better to provide a pure constructor, ideally that cannot error, and then helper methods 

```go
// basic constructor, inject all required deps, ideally without returning errors
func NewRedisCache(client *rediscache.Client, ttl time.Duration) *redisCache

// Provide a helper method for most common use cases
func NewRedisCacheFromURL(connectCtx context.Context, cacheURL string, ttl time.Duration) (*redisCache, error)
```

I haven't changed the names myself to prevent breaking changes